base/slicearray.jl

```diff
@@ -0,0 +1,219 @@
+"""
+    Slices{N,P,SM,AX,S} <: AbstractArray{S,N}
+
+An `AbstractArray` of slices into a parent array.
+
+These should typically be constructed by [`eachslice`](@ref), [`eachcol`](@ref) or
+[`eachrow`](@ref).
+
+[`parent(S::Slices)`](@ref) will return the parent array.
+"""
+struct Slices{N,P,SM,AX,S} <: AbstractArray{S,N}
```

Comment:
@mbauman One other thing I'd appreciate your opinion on is the parameter order: I pushed the element type `S` to the end, since it is typically complicated, and also not that important. I originally had `N` as the first one since it made some constructors easier, but I don't think that is really required any longer.

Reply:
Yeah, I can see this both ways.  I think I'd lean towards just doing the `Slices{S, N, ...}` thing to fit with convention.  It's gonna be a mouthful of a type in any case, might as well match convention.

Reply:
It probably does make sense to have element type after parent type though, as it makes using the `Rows`/`Colums` aliases easier, i.e. `f(X::Rows{Matrix{Float64}}) = ...`